tests/e2e/objects/webviews/DirectConnectionFormWebview.ts

```diff
@@ -0,0 +1,238 @@
+  async fillOtherConnectionType(otherType: string): Promise<void> {
+    await this.otherTypeField.click();
+    await this.page.keyboard.type(otherType);
+  }
+
+  /**
+   * Configure Kafka cluster bootstrap servers.
+   * @param bootstrapServers The bootstrap servers to configure
+   */
+  async fillKafkaBootstrapServers(bootstrapServers: string): Promise<void> {
+    await this.kafkaBootstrapServersField.click();
+    await this.page.keyboard.type(bootstrapServers);
+  }
+
+  /**
+   * Select Kafka authentication type.
+   * @param authType The authentication type to select
+   */
+  async selectKafkaAuthType(authType: SupportedAuthTypes): Promise<void> {
+    await this.kafkaAuthTypeDropdown.click();
+    await this.kafkaAuthTypeDropdown.selectOption(authType);
+  }
+
+  /** Fill credentials for the Kafka configuration. */
+  async fillKafkaCredentials(credentials: Record<string, string>): Promise<void> {
+    for (const [key, value] of Object.entries(credentials)) {
+      const field = this.getCredentialField("kafka_cluster", key);
+      await field.click();
+      await this.page.keyboard.type(value);
+    }
+  }
+
+  /**
+   * Configure Schema Registry URI.
+   * @param uri The Schema Registry URI to configure
+   */
+  async fillSchemaRegistryUri(uri: string): Promise<void> {
+    await this.schemaRegistryUriField.click();
+    await this.page.keyboard.type(uri);
+  }
+
+  /**
+   * Select Schema Registry authentication type.
+   * @param authType The authentication type to select
+   */
+  async selectSchemaRegistryAuthType(authType: SupportedAuthTypes): Promise<void> {
+    await this.schemaRegistryAuthTypeDropdown.click();
+    await this.schemaRegistryAuthTypeDropdown.selectOption(authType);
+  }
+
+  /** Fill credentials for the Schema Registry configuration. */
+  async fillSchemaRegistryCredentials(credentials: Record<string, string>): Promise<void> {
+    for (const [key, value] of Object.entries(credentials)) {
+      const field = this.getCredentialField("schema_registry", key);
+      await field.click();
+      await this.page.keyboard.type(value);
+    }
+  }
+
+  /** Test the connection configuration. */
+  async testConnection(): Promise<void> {
+    await this.testButton.click();
+  }
+
+  /** Save the connection configuration. */
+  async saveConnection(): Promise<void> {
+    await this.saveButton.click();
+  }
+
+  /** Set up a connection using API key/secret for Kafka and/or Schema Registry. */
+  async configureWithApiKeyAndSecret(config: {
+    name: string;
+    connectionType: FormConnectionType;
+    kafka?: {
+      bootstrapServers: string;
+      apiKey: string;
+      apiSecret: string;
+    };
+    schemaRegistry?: {
+      apiKey: string;
+      apiSecret: string;
+      uri: string;
+    };
+  }): Promise<void> {
+    if (!config.kafka && !config.schemaRegistry) {
+      throw new Error("At least one of Kafka or Schema Registry configuration is required");
+    }
```

Comment:
Maybe it worth splitting into two separate methods for kafka and schemaRegistry so that you wouldn't need conditionals at all?

Reply:
I thought about splitting this in half, but figured it would be easier [removing altogether](https://github.com/confluentinc/vscode/pull/2072/commits/a198629c5cbbe5153608777d969330cb0f3ebaff) and using the existing locators and per-field filling methods:
<img width="706" alt="image" src="https://github.com/user-attachments/assets/2eaa53bf-8eca-4cf4-8bc7-f95046bd7cd2" />